File: backend/src/main/java/com/bangchat/data/entity/SpecialAddressing.java
package com.bangchat.data.entity;

import lombok.Data;

/**
 * 角色特殊称呼实体，映射 special_addressing 表。
 * 记录角色对其他角色的专属昵称、叫法和使用备注。
 */
@Data
public class SpecialAddressing {
    private Long id;
    private String characterId;
    private String targetName;
    private String alias;
    private String note;
}

File: bangchat/lib/models/special_addressing.dart
/// 角色特殊称呼实体，映射 special_addressing 表。
class SpecialAddressing {
  final int? id;
  final String characterId;
  final String targetName;
  final String alias;
  final String note;

  const SpecialAddressing({
    this.id,
    required this.characterId,
    required this.targetName,
    required this.alias,
    this.note = '',
  });

  factory SpecialAddressing.fromMap(Map<String, dynamic> map) {
    return SpecialAddressing(
      id: map['id'] as int?,
      characterId: (map['character_id'] as String?) ?? '',
      targetName: (map['target_name'] as String?) ?? '',
      alias: (map['alias'] as String?) ?? '',
      note: (map['note'] as String?) ?? '',
    );
  }
}
